src/functions-and-arrays.js

```diff
@@ -1,41 +1,168 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(numA, numB) {
+  let largestNumber
+
+  if (numA > numB) {
+    largestNumber = numA
+  }
+  else {
+    largestNumber = numB
+  }
+  
+  return largestNumber
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) 
+{
+  if (words.length === 0) { //returns null if array is empty
+    return null
+  }
+
+  if (words.length === 1) {  //returns first word if array has only 1 word
+    return words[0]
+  }
+
+  let longestWord = ""  // returns longest word, first occurence
+  for (let word of words) {
+    if (longestWord.length < word.length) {
+        longestWord = word
+    }
+  }
+  return longestWord
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  if (numbers.length === 0) { //returns 0 if array is empty
+    return 0
+  }
+
+  let sum = 0;  // returns the sum of an array of numbers
+  numbers.forEach(numbers => {
+    sum += numbers
+  });
+  return sum
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(numbers) {
+  if (numbers.length === 0) { //returns 0 if array is empty
+    return 0
+  }
+
+  for (element of numbers) {
+    if (typeof element === "object")
```

Comment:
if you use a different loop just for the throw path then you are going to loop the array twice, you could add it to the other loop